elastic/kibana
# [Discover] Fix data view picker error in aggregation-based Visualizations for saved searches based on adhoc data views

## Summary

Closes #141812

This PR fixes `Search not found` error in data view picker for saed agg visualizations which uses adhoc data views. 
⚠️ Further navigation by clicking `Edit visualization in Lens` will be fixed by #143500


### How to test 

- Go to Discover and create a data view in the data view picker, use "Use without saving" to create an ad-hoc data view
- Save the current search as saved search
- Go to the Visualize Library, select any aggregation-based visualisation
- The visualization is rendered correctly , "Search not found" should not be displayed in the data view picker. `Edit visualization in Lens` should be displyed on top of the page.

**Screenshots (if relevant):**
<img width="1292" alt="Bildschirmfoto 2022-09-26 um 18 14 58" src="https://user-images.githubusercontent.com/463851/192331810-d2652184-5969-4a5b-acde-2e4321bf5f34.png">
